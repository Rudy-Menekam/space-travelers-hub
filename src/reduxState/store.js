import { configureStore } from '@reduxjs/toolkit';
import missionReducer from './missionSlice';
import rocketReducer from './rockets/rocketSlice';

const store = configureStore({
  reducer: {
    missions: missionReducer,
    Rocket: rocketReducer,
  },
});
export default store;
