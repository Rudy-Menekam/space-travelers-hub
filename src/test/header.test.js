import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import Header from "../components/header";
import React from 'react';
import renderer from 'react-test-renderer';
import store from "../reduxState/store";

it('Header page renders correctly', () => {
    const tree = renderer
      .create(
        <React.StrictMode>
          <Provider store={store}>
            <BrowserRouter>
              <Header />
            </BrowserRouter>
          </Provider>
        </React.StrictMode>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });