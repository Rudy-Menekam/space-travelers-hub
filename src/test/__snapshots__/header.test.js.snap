// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header page renders correctly 1`] = `
<nav
  id="nav"
>
  <img
    alt="Planet"
    src="planet.png"
  />
  <h1
    className="logo"
  >
    Space Travelers Hub
  </h1>
  <ul
    className="navlinks"
  >
    <li>
      <a
        aria-current="page"
        className="active"
        href="/"
        onClick={[Function]}
        style={
          Object {
            "textDecoration": "underline",
          }
        }
      >
        Rockets
      </a>
    </li>
    <li>
      <a
        className=""
        href="/missions"
        onClick={[Function]}
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        Missions
      </a>
    </li>
    <div
      className="Line-2"
    />
    <li>
      <a
        className=""
        href="/my_profile"
        onClick={[Function]}
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        My Profile
      </a>
    </li>
  </ul>
</nav>
`;
